packages/core/src/services/documents/resetToVersion.ts

```diff
@@ -0,0 +1,75 @@
+import { and, eq } from 'drizzle-orm'
+import { Commit, DocumentVersion } from '../../browser'
+import { database } from '../../client'
+import {
+  ConflictError,
+  LatitudeError,
+  Result,
+  Transaction,
+  TypedResult,
+} from '../../lib'
+import { documentVersions } from '../../schema'
+import { omit } from 'lodash-es'
+import { findWorkspaceFromCommit } from '../../data-access'
+import { DocumentVersionsRepository } from '../../repositories'
+
+export async function resetToDocumentVersion(
+  {
+    documentVersion,
+    draft,
+  }: {
+    documentVersion: DocumentVersion
+    draft: Commit
+  },
+  db = database,
+): Promise<TypedResult<DocumentVersion, Error>> {
+  const workspace = await findWorkspaceFromCommit(draft, db)
```

Comment:
Pass the workspace instead of making an extra request. You'll we have to do it any way when using in an action